beancount/parser/decimal.h

```diff
@@ -0,0 +1,25 @@
+#ifndef BEANCOUNT_DECIMAL_H
+#define BEANCOUNT_DECIMAL_H
+
+/*
+ * Python does not expose a C-API to create Decimal. This code
+ * implements a system similar to what Python provides for @date and
+ * @datetime objects from the datetime module. The exposed API is
+ * similar to the one implemented (but not exposed) by the decimal
+ * module. Before using any of these functions, the macro
+ * PyDateTime_IMPORT must be invoked.
+ */
+
+static PyObject* decimal_type;
```

Comment:
Hmmm, don't declare variables in a header file; this basically requires that header file to be included in a single compilation unit. You could make it a macro for the declaration, but it would be better to just put that in the .c file.

Reply:
I know this is not optimal. I was trying to keep things simple and at the same time self contained. I'll add a decimal.c for this variable.